src/qt/clientmodel.cpp

```diff
@@ -72,20 +72,28 @@ int ClientModel::getNumBlocks() const
     return chainActive.Height();
 }
 
-int ClientModel::getHeaderTipHeight() const
+int ClientModel::getHeaderTipHeight()
 {
-    LOCK(cs_main);
-    if (!pindexBestHeader)
-        return 0;
-    return pindexBestHeader->nHeight;
+    if (cachedBestHeaderHeight == 0) {
```

Comment:
Where is this initialized? -- these don't have static duration AFAICT.

Also is zero really the best value to use? -- it's in range, so it'll keep relocking again while there is no data there.